orca-queue/src/test/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandlerTest.kt

```diff
@@ -101,7 +104,8 @@ object CompleteStageHandlerTest : SubjectSpek<CompleteStageHandler>({
   val publisher: ApplicationEventPublisher = mock()
   val exceptionHandler: ExceptionHandler = DefaultExceptionHandler()
   val clock = fixedClock()
-  val registry = NoopRegistry()
+  val metricId: Id = mock()
+  val registry = spy(NoopRegistry())
```

Comment:
Actually, as I look at this more closely, would it be simpler to change this to DefaultRegistry?  I think there's no longer a need to mock metricId.  The assertions change to something like (untested):
```
    assertThat(registry.timers()).hasSize(1);
    Timer timer = registry.timers().findFirst().orElseThrow(AssertionError::new);
    assertThat(timer.id().name()).isEqualTo("stage.invocations.duration");
    assertThat(timer.id().tags())
        .contains(
            tag("tag1", "value1"),
            tag("tag2", "value2"));
```

Reply:
Yea, using DefaultRegistry does simplify things. I have modified that and removed the metricId mock, however in a different way from the snippet above, as `timers` have more than one element, and we cannot assume it has size=1.
I modified to validate the timer using `argumentCaptor` as below:
```
            argumentCaptor<Id>().let {
              verify(registry, atLeastOnce()).timer(it.capture())
              val metricId = it.firstValue
              assertThat(metricId.name()).isEqualTo("stage.invocations.duration")
              assertThat(metricId.tags().any{ t -> t.key() == "tag1" && t.value() == "value1" }).isTrue
              assertThat(metricId.tags().any{ t -> t.key() == "tag2" && t.value() == "value2" }).isTrue
            }
```